nodejs/aws-infra/iam/index.ts

```diff
@@ -0,0 +1,213 @@
+     */
+    readonly name?: pulumi.Input<string>;
+
+    /**
+     * Path in which to create the user.
+     */
+    readonly path?: pulumi.Input<string>;
+
+    /**
+     * The ARN of the policy that is used to set the permissions boundary for the user.
+     */
+    readonly permissionsBoundary?: pulumi.Input<string>;
+
+    /**
+     * Key-value mapping of tags for the IAM user
+     */
+    readonly tags?: pulumi.Input<{
+        [key: string]: any;
+    }>;
+
+    /**
+     * An existing User to use for this awsinfra User.  If not provided, a default one will be
+     * created.
+     */
+    user?: aws.iam.User;
+}
+
+// Make sure our exported args shape is compatible with the overwrite shape we're trying to provide.
+const test1: string = utils.checkCompat<OverwriteUserArgs, UserArgs>();
+
+type OverwriteUserGroupMembershipArgs = utils.Overwrite<
+    aws.iam.UserGroupMembershipArgs,
+    { user?: never }
+>;
+
+/**
+ * The set of arguments for constructing a UserGroupMembership resource.
+ */
+export interface UserGroupMembershipArgs {
+    /**
+     * A list of [IAM Groups][1] to add the user to
+     */
+    readonly groups: pulumi.Input<pulumi.Input<string>[]>;
+}
+
+// Make sure our exported args shape is compatible with the overwrite shape we're trying to provide.
+const test2: string = utils.checkCompat<
+    OverwriteUserGroupMembershipArgs,
+    UserGroupMembershipArgs
+>();
+
+type OverwriteAccessKeyArgs = utils.Overwrite<aws.iam.AccessKeyArgs, { user?: never }>;
+
+/**
+ * The set of arguments for constructing a AccessKey resource.
+ */
+export interface AccessKeyArgs {
+    /**
+     * Either a base-64 encoded PGP public key, or a
+     * keybase username in the form `keybase:some_person_that_exists`.
+     */
+    readonly pgpKey?: pulumi.Input<string>;
+}
+
+// Make sure our exported args shape is compatible with the overwrite shape we're trying to provide.
+const test3: string = utils.checkCompat<OverwriteAccessKeyArgs, AccessKeyArgs>();
+
+/**
+ * EmployeeUser represents an AWS IAM User that is a full-time employee. Because it is useful to
+ * quickly distinguish between various types of users, all EmployeeUsers are allocated names of the
+ * form `employee.${name}`.
+ */
+export class EmployeeUser extends User {
+    /**
+     * Create a EmployeeUser resource with the given unique name, arguments, and options.
+     *
+     * @param name The _unique_ name of the resource. This name will be transformed into the form
+     * `employee.${name}`.
+     * @param args The arguments to use to populate this resource's properties.
+     * @param opts A bag of options that control this resource's behavior.
+     */
+    constructor(name: string, args?: UserArgs, opts?: pulumi.CustomResourceOptions) {
+        super(`employee.${name}`, args, opts);
+    }
+}
```

Comment:
is this for testing purposes?

Reply:
As mentioned offline:

1. this is is a broadly-accepted best practice ([_cf._](https://github.com/gruntwork-io/module-security-public/tree/master/modules/iam-policies#iam-users)), which
2. we will be pushing up into the "prod" packages, and pulling it back into x when we're sure it's the right thing to do.